### PR TITLE
Refactor adding job defaults out of BaseJob and into Utils

### PR DESCRIPTION
**Pull request by imuchnik**
That makes it independently available to other builders.

### CODE REVIEW

**Comment by vickumar1981** [comment]
:+1:

File: src/main/groovy/jenkins/automation/utils/CommonUtils.groovy [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * @param context  delegate passed in context\
+ */
+class CommonUtils {
+    static void add_defaults(context) {
+        context.with {
+            wrappers {
+                colorizeOutput()
+            }
+            logRotator {
+                numToKeep(10)
```
Comment (marcesher) [inline src/main/groovy/jenkins/automation/utils/CommonUtils.groovy]: I think 10 is too low. Let's change the default to something more reasonable and in line with how a lot of our jobs are currently configured.... I'd say at least 30

File: src/main/groovy/jenkins/automation/builders/BaseJobBuilder.groovy [hunk 2]
```diff
@@ -25,46 +26,18 @@ class BaseJobBuilder {
...
-            }
-            logRotator {
-                //TODO: make it parametrized to builder
-                numToKeep(10)
-            }
-            publishers{
-                allowBrokenBuildClaiming()
-
-                if(emails){
+            CommonUtils.add_defaults(delegate)
```
Comment (marcesher) [inline src/main/groovy/jenkins/automation/builders/BaseJobBuilder.groovy]: please change this to `addDefaults`, to match the style of the rest of the code

Comment (imuchnik) [inline src/main/groovy/jenkins/automation/builders/BaseJobBuilder.groovy]: should we allow overrides?

Comment (marcesher) [inline src/main/groovy/jenkins/automation/builders/BaseJobBuilder.groovy]: Well... that means that every single builder will also need to support that parameter. Is it worth it, compared with just adding a new block in your jobs.groovy for that job in the cases where you do need to overwrite it?

Comment (imuchnik) [inline src/main/groovy/jenkins/automation/builders/BaseJobBuilder.groovy]: we can leave it with a sensible default and optional ability to override. Net total effect is the same with configure block, but they achieve it with a one-liner parameter.

**Comment by marcesher** [comment]
@imuchnik thanks!

**Pull request merged**